RegressionModels/LinearRegression/HousePriceEstimation.py

```diff
@@ -0,0 +1,174 @@
+labels = Data_updated['Exterior2nd'].astype('category').cat.categories.tolist()
+replace_map_comp = {'Exterior2nd' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['MasVnrType'].astype('category').cat.categories.tolist()
+replace_map_comp = {'MasVnrType' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['ExterQual'].astype('category').cat.categories.tolist()
+replace_map_comp = {'ExterQual' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['Foundation'].astype('category').cat.categories.tolist()
+replace_map_comp = {'Foundation' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['BsmtQual'].astype('category').cat.categories.tolist()
+replace_map_comp = {'BsmtQual' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['BsmtExposure'].astype('category').cat.categories.tolist()
+replace_map_comp = {'BsmtExposure' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['BsmtFinType1'].astype('category').cat.categories.tolist()
+replace_map_comp = {'BsmtFinType1' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['HeatingQC'].astype('category').cat.categories.tolist()
+replace_map_comp = {'HeatingQC' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['GarageType'].astype('category').cat.categories.tolist()
+replace_map_comp = {'GarageType' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['GarageFinish'].astype('category').cat.categories.tolist()
+replace_map_comp = {'GarageFinish' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+labels = Data_updated['KitchenQual'].astype('category').cat.categories.tolist()
+replace_map_comp = {'KitchenQual' : {k: v for k,v in zip(labels,list(range(1,len(labels)+1)))}}
+print(replace_map_comp)
+
+Data_updated_replace.replace(replace_map_comp, inplace=True)
+
+Data_updated_replace = Data_updated_replace.dropna()
+
```

Comment:
create a function which can convert categorical columns to their respective numerical representations and map it on all categorical columns in single go